R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,266 @@
+#' Derive a Clinical Benefit Parameter
+#'
+#' Add a clinical benefit/disease control parameter to the input dataset.
+#'
+#' @details
+#' Clinical benefit/disease control is first identified for looking for subjects
+#' having response status, and then derived for subjects that have at least one
+#' evaluable non-PD response assessment prior to first PD (Progressive Disease)
+#' (i.e., inclusive of `CR`, `PR`, `SD` and exclusive of `NA`, `NE`, `ND`, and
+#' `PD`) and after a specified amount of time from a reference date (`ref_start_window`).
+#'
+#' \enumerate{
+#'   \item The input dataset (`dataset`) is restricted to the observations matching
+#'   `filter_source` and to observations before or at the date specified by
+#'   `source_pd`.
+#'
+#'   \item Variables `AVALC`, `AVAL`, and `ADT` will be populated for each unique subject
+#'   in the input `dataset_adsl` as identified by `subject_keys`.
```

Comment:
I do not understand this step. `AVALC`, `AVAL`, and `ADT` are described below.

Reply:
this was in attempt to address https://github.com/pharmaverse/admiralonco/pull/40#discussion_r894461001

Reply:
Hm, do you want to emphasize that a new observation is created for each subject?

Reply:
rephrased